lib/listen/listener.rb

```diff
@@ -82,42 +79,39 @@ def initialize(*dirs, &block)
       super() # FSM
     end
 
-    default_state :initializing
+    start_state :initializing
 
     state :initializing, to: [:backend_started, :stopped]
 
-    state :backend_started, to: [:frontend_ready, :stopped] do
+    state :backend_started, to: [:processing_events, :stopped] do
       backend.start
     end
 
-    state :frontend_ready, to: [:processing_events, :stopped] do
-      processor.setup
-    end
-
     state :processing_events, to: [:paused, :stopped] do
-      processor.resume
+      processor.start
     end
 
     state :paused, to: [:processing_events, :stopped] do
       processor.pause
     end
 
     state :stopped, to: [:backend_started] do
-      backend.stop # should be before processor.teardown to halt events ASAP
-      processor.teardown
+      backend.stop # should be before processor.stop to halt events ASAP
+      processor.stop
     end
 
     # Starts processing events and starts adapters
     # or resumes invoking callbacks if paused
     def start
-      Listen::Logger.debug("InvocaDebug: Listener#start about to transition :backend_started state: #{state}")
-      transition :backend_started if state == :initializing
-      Listen::Logger.debug("InvocaDebug: Listener#start about to transition :frontend_ready state: #{state}")
-      transition :frontend_ready if state == :backend_started
-      Listen::Logger.debug("InvocaDebug: Listener#start about to transition :processing_events state: #{state}")
-      transition :processing_events if state == :frontend_ready
-      Listen::Logger.debug("InvocaDebug: Listener#start about to transition :processing_events state: #{state}")
-      transition :processing_events if state == :paused
+      case state
```

Comment:
There were only two supported cases. After I puzzled that out from the old code, I saw the comment saying exactly this!